lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1559,8 +1559,8 @@ def test_progress_queries_paced_courses(self, self_paced, query_count):
 
     @patch.dict(settings.FEATURES, {'ASSUME_ZERO_GRADE_IF_ABSENT_FOR_ALL_TESTS': False})
     @ddt.data(
-        (False, 63, 46),
-        (True, 55, 40)
+        (False, 63, 44),
+        (True, 55, 38)
```

Comment:
I have no idea! It snuck in during my rebase.

Should I dig in to try to figure out why?

Reply:
Ok, I'm really confused. 
[Tests were passing with this change](https://github.com/edx/edx-platform/pull/28440#issuecomment-921659513), but when I dug in on my devstack to try to figure out why the query count was different, they started failing.

So I've reverted this change (71106bd64fa9cc8ee0e4b438caed9016aca75ea2), and will see what the testing says now.

Reply:
Yes, let's make sure we understand this.

Reply:
Sure, I'll keep digging.

But I'm really struggling to reproduce these results on my devstack. When I run the `test_progress_queries` tests there, the counts from `master` work, but [they don't work when Jenkins runs them](https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/34669/). This indicates some difference between the test settings for the Jenkins tests that aren't being set in the devstack, maybe a waffle switch somewhere?

These are the 2 missing queries, which are expected to be executed by the subsequent calls to render the progress page, but aren't being called when Jenkins runs these tests:

```sql
SELECT "schedules_schedule"."id", "schedules_schedule"."created", "schedules_schedule"."modified", "schedules_schedule"."enrollment_id", "schedules_schedule"."active", "schedules_schedule"."start_date", "schedules_schedule"."upgrade_deadline" FROM "schedules_schedule" INNER JOIN "student_courseenrollment" ON ("schedules_schedule"."enrollment_id" = "student_courseenrollment"."id") WHERE ("student_courseenrollment"."course_id" = 'org.12/course_12/Run_12' AND "student_courseenrollment"."user_id" = 2)

SELECT "edx_when_userdate"."id", "edx_when_userdate"."created", "edx_when_userdate"."modified", "edx_when_userdate"."user_id", "edx_when_userdate"."content_date_id", "edx_when_userdate"."abs_date", "edx_when_userdate"."rel_date", "edx_when_userdate"."reason", "edx_when_userdate"."actor_id", "edx_when_contentdate"."id", "edx_when_contentdate"."course_id", "edx_when_contentdate"."policy_id", "edx_when_contentdate"."location", "edx_when_contentdate"."field", "edx_when_contentdate"."active", "edx_when_datepolicy"."id", "edx_when_datepolicy"."created", "edx_when_datepolicy"."modified", "edx_when_datepolicy"."abs_date", "edx_when_datepolicy"."rel_date" FROM "edx_when_userdate" INNER JOIN "edx_when_contentdate" ON ("edx_when_userdate"."content_date_id" = "edx_when_contentdate"."id") INNER JOIN "edx_when_datepolicy" ON ("edx_when_contentdate"."policy_id" = "edx_when_datepolicy"."id") WHERE ("edx_when_contentdate"."active" = 1 AND "edx_when_contentdate"."course_id" = 'org.12/course_12/Run_12' AND "edx_when_userdate"."user_id" = 2) ORDER BY "edx_when_userdate"."modified" ASC
```

I'm not familiar with these tables, so need to dig through the codebase to figure out how where they're being used on the progress page.

Reply:
Oh man, this was a pain to find!

It was caused by updating [edx-when from 2.1.0 to 2.2.0](https://github.com/edx/edx-platform/pull/28440/files#diff-160eb30f60d519b8fe8a919cb2646a7ebec7652952129cfdd803776e70be6300L509), I likely because of [the efficiencies introduced by this change](https://github.com/edx/edx-when/commit/b9930ec7cc3e8c884113bffc988c07c95a58fe5f).

https://github.com/edx/edx-when/compare/2.1.0...2.2.0

Reply:
Got it. I am assuming the dependencies change will go away once you rebase?

Reply:
Ah yep -- because they've since [constrained `edx-when==2.1.0`](https://github.com/edx/edx-platform/blob/master/requirements/constraints.txt#L111).

But I still need to update the requirements to include the change from [XBlock#542](https://github.com/edx/XBlock/pull/542). It doesn't look like the tag you pushed generated [a new release](https://github.com/edx/XBlock/releases)? It should be `1.5.1`, not [v1.5.1](https://github.com/edx/XBlock/tree/v1.5.1). Could you push a new tag please?

Reply:
Sorry about that. I'm going to merge the 2.2.2 version of edx-when today, but I'll hold off on the version bump (and subsequent query adjustment) until after this one lands.

Reply:
Sure, i have pushed [1.5.1](https://github.com/edx/XBlock/tree/1.5.1).

Reply:
@symbolist Hmm.. that still hasn't triggered a release, and that's what we need here: https://github.com/edx/XBlock/releases

Could you work with your core committer champions on XBlock to sort this out?